run.go

```diff
@@ -1924,9 +1924,21 @@ func runCopyStdio(stdio *sync.WaitGroup, copyPipes bool, stdioPipe [][]int, copy
 				return true
 			}
 		}
-		logrus.Errorf("%s: %v", what, err)
+		logrus.Error(what)
 		return false
 	}
+	logIfUnexpectedWhileDraining := func(err error, what string) {
+		if err == nil {
+			return
+		}
+		if errno, isErrno := err.(syscall.Errno); isErrno {
+			switch errno {
+			case syscall.EIO:
+				return
+			}
+		}
+		logIfNotRetryable(err, what)
+	}
```

Comment:
To reduce the cycl. complexity, we could refactor that into a single function that can be called here and in `chroot/run.go`.